src/Model/Widget.php

```diff
@@ -267,10 +252,13 @@ public function getController()
         }
 
         if (!class_exists($controllerClass)) {
-            throw new Exception('Could not find controller class for ' . $controllerClass);
+            throw new Exception('Could not find controller class for ' . static::class);
```

Comment:
The idea is to say "we couldn't find a controller for MyCustomWidget"

Reply:
Right, maybe it's me getting confused between `self::class`, `static::class`, `get_class()`, and `__CLASS__`, but I feel like the difference here is that `static::class` will be _this_ class, i.e. `Widget` or a subclass there of.

Where as the variable here being used in the `if` condition will be for a corresponding controller, a subclass of `WidgetController`.

tl;dr, I think this error message will give the wrong information.

Reply:
Perhaps it needs a better doc block.

What the method is doing is trying to find an appropriate controller for the current Widget instance. It does this by looking backwards through the inheritance chain until it finds a controller class that exists.

This error is thrown at the end when it couldn't find any controllers at all, and the `$controllerClass` at this point will be `SilverStripe\View\ViewableDataController` which is misleading and incorrect.

I changed it to include the late static bound (`static::class`) class name of the widget that threw the error, so that when you use the CMS the error you get will say "we couldn't find a controller for YourCustomWidgetName", rather than some low level class that has nothing to do with the widgets in this context.

Does this make sense?

Reply:
Yep, perfect sense :)
Thanks for explaining that 👍